### PR TITLE
Fix #84 Wrong MCC+MNC encoding

### LINKED ISSUE #84: [BUG] Three-digit MNCs are encoded incorrectly
When configuring a 3-digit MNC in the config.yml, PacketRusher does not encode it correctly. 

**configuration in config.yml**

> gnodeb:
>   controlif:
>     ip: "xx.yy.zz.ww"
>     port: 9487
>   dataif:
>     ip: "xx.yy.zz.uu"
>     port: 2152
>   plmnlist:
>     mcc: "789"
>     **mnc: "123"**
> ...

**encoding in the NGSetupRequest message**

```
> NG Application Protocol (NGSetupRequest)
>     NGAP-PDU: initiatingMessage (0)
>         initiatingMessage
>             procedureCode: id-NGSetup (21)
>             criticality: reject (0)
>             value
>                 NGSetupRequest
>                     protocolIEs: 4 items
>                         Item 0: id-GlobalRANNodeID
>                             ProtocolIE-Field
>                                 id: id-GlobalRANNodeID (27)
>                                 criticality: reject (0)
>                                 value
>                                     GlobalRANNodeID: globalGNB-ID (0)
>                                         globalGNB-ID
>                                             pLMNIdentity: 873921
>                                                 Mobile Country Code (MCC): Unassigned (789)
>                                                 Mobile Network Code (MNC): Unknown (312)
>                                             gNB-ID: gNB-ID (0)
>                                                 gNB-ID: 0f423d [bit length 24, 0000 1111  0100 0010  0011 1101 decimal value 999997]
>                         Item 1: id-RANNodeName
>                             ProtocolIE-Field
>                                 id: id-RANNodeName (82)
>                                 criticality: ignore (1)
>                                 value
>                                     RANNodeName: PacketRusher

You can see in the NGSetupRequest message that the MNC incorrectly becomes "312" instead of "123".

### PR DESCRIPTION
**Pull request by linouxis9**
## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

- [X] I have read the **CONTRIBUTING** document.
- [X] Each of my commits messages include `Signed-off-by: Author Name <authoremail@example.com>` to accept the DCO.

### CODE REVIEW

**Comment by linouxis9** [comment]
Hi @rjaksa,

Your bug should be fixed by this PR. Can you try PacketRusher again with this fix / using branch `fixmnc`.

Thanks a lot,
Valentin

File: internal/control_test_engine/gnb/context/context.go [hunk 1]
```diff
@@ -422,29 +422,19 @@ func (gnb *GNBContext) GetMccAndMnc() (string, string) {
 }
 
 func (gnb *GNBContext) GetMccAndMncInOctets() []byte {
```
Comment (Raguideau) [inline internal/control_test_engine/gnb/context/context.go:424]: GetMccAndMncInOctets() are the same for UE and GNB, woudn't it be better to make a common function?

File: internal/utils/pcap.go
Comment (Raguideau) [inline internal/utils/pcap.go:1]: File deleted by mistake?

**Comment by rjaksa** [comment]
The code change works for the gNB. But there is an informative printf regarding the PLMNs Identities supported by AMF that shows the MNC incorrectly. It lists the 3-digit MNC with the digits in encoded order, instead of the "human readable" format. E.g. MNC 123 is printed as "231".

E.g. for MCC=987, MNC=123, the printf looks like:

> INFO[0000] [GNB][AMF] PLMNs Identities Supported by AMF -- mcc: 987 mnc:231

**Comment by linouxis9** [comment]
Thank you @rjaksa, it's fixed!

**Pull request merged**